swap/swap.go

```diff
@@ -609,21 +624,14 @@ func (s *Swap) StartChequebook(chequebookAddrFlag common.Address, initialDeposit
 	}
 	// nothing written to state disk before, no flag provided: deploying new chequebook
 	if err == state.ErrNotFound && chequebookAddrFlag == (common.Address{}) {
-		var toDeposit = initialDepositAmount
-		if toDeposit == 0 {
-			toDeposit, err = promptInitialDepositAmount()
-			if err != nil {
-				return nil, err
-			}
-		}
-		if contract, err = s.Deploy(context.TODO(), toDeposit); err != nil {
+
+		if contract, err = s.Deploy(context.TODO()); err != nil {
```

Comment:
I would put a `context.Background()` here.

Reply:
Done, can you explain to me when we would choose the one over the other? 
As far as I got it, we need to change all those contexts at one point to a more sensible value and both `context.TODO()` as well as `context.Background()` are some kind of placeholders, to be used until we put a better value there.

Reply:
Considering this is during startup and there is no point in continuing if the transaction has not succeeded yet `context.Background` should be fine here. It means we are waiting until either the transaction has succeeded or failed or the node is shutting down.